### ISSUE TITLE
Gamemode files watcher crashes when editing a file

### ISSUE THREAD

**Issue by Deewarz** [MafiaHub/Framework#79]
## How to:
- Run the server
- Edit `gamemode/index.js` and save

## Additionnal information
Partial error log:
```
[17:45:34] [debug] [Scripting] Gamemode is reloaded due to the file changes
FATAL ERROR: v8::ToLocalChecked Empty MaybeLocal.
 1: 00007FF8BBE2850F CRYPTO_memcmp+794783
 2: 00007FF8BBEF6687 CRYPTO_memcmp+1638935
 3: 00007FF8BBEF8052 CRYPTO_memcmp+1645538
 4: 00007FF8BC50DF95 CRYPTO_memcmp+8026917
 5: 00007FF8BBE43E7C CRYPTO_memcmp+907788
 6: 00007FF8BBE5A6DF CRYPTO_memcmp+1000047
 7: 00007FF8BBE3D185 CRYPTO_memcmp+879893
 8: 00007FF8BBDEE1E1 CRYPTO_memcmp+556401
 9: 00007FF7744E4C64 Framework::Scripting::Engines::Node::Engine::LoadGamemode+2004 [Framework\code\framework\src\scripting\engines\node\engine.cpp]:L221
10: 00007FF7744E6173 `Framework::Scripting::Engines::Node::Engine::WatchGamemodeChanges'::`2'::<lambda_1>::operator()+355 [Framework\code\framework\src\scripting\engines\node\engine.cpp]:L332
```

**Comment by zpl-zak** [comment]
Additional logs from a recent test:
```
FATAL ERROR: v8::ToLocalChecked Empty MaybeLocal.
 1: 00007FFD6A44850F node::DumpBacktrace+143 [C:\Projects\node-v18.12.1\src\debug_utils.cc]:L310
 2: 00007FFD6A516687 node::Abort+23 [C:\Projects\node-v18.12.1\src\node_errors.cc]:L298
 3: 00007FFD6A518052 node::OnFatalError+274 [C:\Projects\node-v18.12.1\src\node_errors.cc]:L495
 4: 00007FFD6AB2DF95 v8::api_internal::ToLocalEmpty+53 [C:\Projects\node-v18.12.1\deps\v8\src\api\api.cc]:L904
 5: 00007FFD6A463E7C node::Environment::CreateProperties+380 [C:\Projects\node-v18.12.1\src\env.cc]:L598
 6: 00007FFD6A47A6DF node::Environment::InitializeMainContext+303 [C:\Projects\node-v18.12.1\src\env.cc]:L777
 7: 00007FFD6A45D185 node::Environment::Environment+133 [C:\Projects\node-v18.12.1\src\env.cc]:L765
 8: 00007FFD6A40E1E1 node::CreateEnvironment+129 [C:\Projects\node-v18.12.1\src\api\environment.cc]:L360
 9: 00007FF626D60954 Framework::Scripting::Engines::Node::Engine::LoadGamemode+2004 [E:\Framework\code\framework\src\scripting\engines\node\engine.cpp]:L233
10: 00007FF626D61E63 `Framework::Scripting::Engines::Node::Engine::WatchGamemodeChanges'::`2'::<lambda_1>::operator()+355 [E:\Framework\code\framework\src\scripting\engines\node\engine.cpp]:L343
11: 00007FF626BD9CEA std::invoke<`Framework::Scripting::Engines::Node::Engine::WatchGamemodeChanges'::`2'::<lambda_1> &,cppfs::FileHandle &,enum cppfs::FileEvent>+42 [C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.39.33218\include\type_traits]:L1754
12: 00007FF626D6212E std::_Func_impl_no_alloc<`Framework::Scripting::Engines::Node::Engine::WatchGamemodeChanges'::`2'::<lambda_1>,void,cppfs::FileHandle &,enum cppfs::FileEvent>::_Do_call+46 [C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.39.33218\include\functional]:L909
13: 00007FF626E6A526 std::_Func_class<void,cppfs::FileHandle &,enum cppfs::FileEvent>::operator()+118 [C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.39.33218\include\functional]:L952
14: 00007FF626E6A3C6 cppfs::FunctionalFileEventHandler::onFileEvent+70 [E:\Framework\vendors\cppfs\source\cppfs\source\FunctionalFileEventHandler.cpp]:L28
15: 00007FF626E5E3A2 cppfs::FileWatcher::onFileEvent+130 [E:\Framework\vendors\cppfs\source\cppfs\source\FileWatcher.cpp]:L127
16: 00007FF626E763B5 cppfs::AbstractFileWatcherBackend::onFileEvent+53 [E:\Framework\vendors\cppfs\source\cppfs\source\AbstractFileWatcherBackend.cpp]:L23
17: 00007FF626E719C9 cppfs::LocalFileWatcher::watch+1849 [E:\Framework\vendors\cppfs\source\cppfs\source\windows\LocalFileWatcher.cpp]:L234
18: 00007FF626E5E311 cppfs::FileWatcher::watch+81 [E:\Framework\vendors\cppfs\source\cppfs\source\FileWatcher.cpp]:L120
19: 00007FF626D5FF6D Framework::Scripting::Engines::Node::Engine::Update+349 [E:\Framework\code\framework\src\scripting\engines\node\engine.cpp]:L134
20: 00007FF626D5E22F Framework::Scripting::Module::Update+47 [E:\Framework\code\framework\src\scripting\module.cpp]:L74
21: 00007FF626CDD6B3 Framework::Integrations::Server::Instance::Update+259 [E:\Framework\code\framework\src\integrations\server\instance.cpp]:L357
22: 00007FF626CDD7E0 Framework::Integrations::Server::Instance::Run+32 [E:\Framework\code\framework\src\integrations\server\instance.cpp]:L372
23: 00007FF626BC2C4D main+445 [E:\Framework\code\projects\MafiaMP\code\server\src\main.cpp]:L34
24: 00007FF6270FEC0C __scrt_common_main_seh+268 [D:\a\_work\1\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl]:L288
25: 00007FFE7238257D BaseThreadInitThunk+29
26: 00007FFE73A6AA58 RtlUserThreadStart+40
```

**Comment by zpl-zak** [comment]
Crashes here, in node's env.cc file:
![devenv_qpNzzL7wmj](https://github.com/MafiaHub/Framework/assets/9026786/7d78fbec-f1cc-4540-9506-2ef2d5945046)

**Comment by zpl-zak** [comment]
Fixed by #84 

![WindowsTerminal_zRZq1Tmf0K](https://github.com/MafiaHub/Framework/assets/9026786/25a71bbe-31fb-4172-8ba2-3802e9e9695c)
